### PR TITLE
Custom Views context hook adjustment

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Custom Views context hook adjustment

#### Description

In this PR I propose refactoring the `useCustomView` hook to not only return the Custom Views context but also the internal Application context. This way, Custom Views developers requirement some data from any of the context can access to them from the same hook (dedicated to Custom Views) without them needing to know we manage two context internally.

### CODE REVIEW

File: packages/application-shell/src/test-utils/custom-views-utils.tsx [hunk 1]
```diff
@@ -45,6 +49,10 @@ export const renderCustomView = (
       locale: props.locale,
       project: {
         key: props.projectKey,
+        allAppliedPermissions: props.projectAllAppliedPermissions || [],
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/test-utils/custom-views-utils.tsx:52]: Writing the docs I found out we allow for custom permissions in Custom Application testing so I'm adding that possibility for Custom Views as well.

File: packages/application-shell/src/test-utils/custom-views-utils.tsx [hunk 2]
```diff
@@ -45,6 +49,10 @@ export const renderCustomView = (
...
+      },
+      environment: {
+        entryPointUriPath: CUSTOM_VIEW_HOST_ENTRY_POINT_URI_PATH,
```
Comment (emmenko) [inline packages/application-shell/src/test-utils/custom-views-utils.tsx:55]: 💯

File: packages/application-shell-connectors/src/components/custom-view-context/custom-view-context.tsx [hunk 3]
```diff
@@ -1,11 +1,17 @@
 import { createContext, useContext, useMemo, type ReactNode } from 'react';
 import type { CustomViewData } from '@commercetools-frontend/constants';
+import {
+  type TApplicationContext,
+  useApplicationContext,
+} from '../application-context';
 
 export type TCustomViewContext = {
   hostUrl: string;
   config: CustomViewData;
```
Comment (emmenko) [inline packages/application-shell-connectors/src/components/custom-view-context/custom-view-context.tsx:10]: Nit: now that we have the data together, should we rename this to `customViewConfig`?

**Review by emmenko — APPROVED** [review]
Nice, thanks for following up on this! 🤗

PS: changeset

File: packages/application-shell/src/components/custom-view-shell/custom-view-shell.tsx [hunk 4]
```diff
@@ -44,6 +45,7 @@ type THostEventData = {
 };
 
 type TCustomViewShellProps = {
+  apolloClient?: ApolloClient<NormalizedCacheObject>;
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/components/custom-view-shell/custom-view-shell.tsx:48]: Also allow consumers to provide their custom Apollo client (same as with Custom Applications)

File: packages/application-shell-connectors/src/components/custom-view-context/custom-view-context.tsx (hunk 3 above)
Comment (CarlosCortizasCT) [reply packages/application-shell-connectors/src/components/custom-view-context/custom-view-context.tsx]: Yes, that makes sense. Thanks 👍 

Updated here: fa5fa67133a501c01858a1bde3737e9387e8574d

**Review by emmenko — APPROVED** [review]

File: packages/application-config/src/formatters.ts [hunk 5]
```diff
@@ -118,6 +122,15 @@ function entryPointUriPathToResourceAccesses<
   };
 }
 
+function resolveCustomViewResourceAccesses<PermissionGroupName extends string>(
```
Comment (CarlosCortizasCT) [inline packages/application-config/src/formatters.ts:125]: When writing the docs for Custom Views I noticed if a developer needs to calculate the permissions he would need to use something like this:
```js
export const PERMISSIONS = entryPointUriPathToPermissionKeys(CUSTOM_VIEW_HOST_ENTRY_POINT_URI_PATH);
```

It seems confusing to me as the concept of `entry point` does not exist for Custom Views, so I decided to propose the inclusion of this new helper target for Custom Views where they can get both the basic resource accesses and also the one for the additional groups.

File: packages/application-shell/src/test-utils/custom-views-utils.tsx [hunk 6]
```diff
@@ -23,8 +30,12 @@ const testCustomViewData: CustomViewData = {
 type TRenderCustomViewParams = {
   locale: string;
   projectKey?: string;
+  projectAllAppliedPermissions?: { name: string; value: boolean }[];
   customViewHostUrl?: string;
   customViewConfig?: Partial<CustomViewData>;
+  apolloClient?: ApolloClient<NormalizedCacheObject>;
+  environment?: Partial<TProviderProps<{}>['environment']>;
+  user?: Partial<TProviderProps<{}>['user']>;
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/test-utils/custom-views-utils.tsx:38]: I'm also adding the `environment` and the `user` as they also seem quite helpful when testing Custom Views with different permissions.

File: packages/application-config/src/formatters.ts (hunk 5 above)
Comment (CarlosCortizasCT) [reply packages/application-config/src/formatters.ts:125]: @emmenko please let me knot what you think about this 🙏

Comment (CarlosCortizasCT) [reply packages/application-config/src/formatters.ts]: Thanks for the suggestion.

I updated this part here: 96cc64e73e425db269f70c601169871eed7eff6f

File: .changeset/cuddly-clouds-pull.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/application-config': minor
+---
+
+Introducing the new `resolveCustomViewResourceAccesses` helper which allows calculating the resource accesses for a Custom View.
```
Comment (emmenko) [inline .changeset/cuddly-clouds-pull.md:5]:
```suggestion
Add new helper functions `computeCustomViewPermissionsKeys` and `computeCustomViewResourceAccesses` to compute the permission names for Custom Views.
```

**Review by emmenko — APPROVED** [review]

**Pull request merged**